data/UnitOfWork.js

```diff
@@ -0,0 +1,27 @@
+const TaskMemoryDAO = require('./TaskMemoryDAO');
+const TaskRepository = require('./TaskRepository');
+const CourseMemoryDAO = require('./CourseMemoryDAO');
+const CourseRepository = require('./CourseRepository');
+
+const validateTaskTitle = (title) => (title && (typeof title === 'string' || title instanceof String));
+const validateCourseId = (id) => (id && Number.isInteger(id));
```

Comment:
You don't need the falsy check for this because `Number.isInteger` will be false if it's null, undefined, or an empty string. This can just be

```javascript
const validateCourse = id => Number.isInteger(id);
```